src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -188,3 +228,54 @@ You can usually call `app.render(request)` without using `.match` because Astro
 ```
 
 Once you [publish your adapter to npm](https://docs.npmjs.com/cli/v8/commands/npm-publish), running `astro add example` will install your package with any peer dependencies specified in your `package.json`. We will also instruct users to update their project config manually.
+
+
+## Astro features
+
+<Since v="3.0.0" />
+
+Astro features are a way for an adapter to tell Astro if they are able to support a feature - and its level of support.  
+
+When using these properties, Astro will:
+- run specific validation; 
+- emit contextual to the logs;
+
+These operations are run based on the features supported or not supported, their level of support, and the configuration that the user uses.
+
+```js title="my-adapter.mjs" ins={9-15}
+export default function createIntegration() {
+  return {
+    name: '@matthewp/my-adapter',
+    hooks: {
+      'astro:config:done': ({ setAdapter }) => {
+        setAdapter({
+          name: '@matthewp/my-adapter',
+          serverEntrypoint: '@matthewp/my-adapter/server.js',
+          supportedAstroFeatures: {
+            assets: {
+              supportKind: "experimental",
+	            isSharpCompatible: false,
+	            isSquooshCompatible: false
+            }
+          }  
+        });
+      },
+    },
+  };
+}
+```
+```
+
+In this example, we're telling Astro that this adapter has experimental support for assets, but it's not compatible with the built-in services.
```

Comment:
``````suggestion
The following configuration tells Astro that this adapter has experimental support for assets, but the adapter is not compatible with the built-in services Sharp and Squoosh:

```js title="my-adapter.mjs" ins={9-15}
export default function createIntegration() {
  return {
    name: '@matthewp/my-adapter',
    hooks: {
      'astro:config:done': ({ setAdapter }) => {
        setAdapter({
          name: '@matthewp/my-adapter',
          serverEntrypoint: '@matthewp/my-adapter/server.js',
          supportedAstroFeatures: {
            assets: {
              supportKind: "experimental",
	            isSharpCompatible: false,
	            isSquooshCompatible: false
            }
          }  
        });
      },
    },
  };
}
```
``````
I just changed the order around, so people know what to look for in the code sample first!